activerecord/lib/active_record/associations/association.rb

```diff
@@ -229,14 +229,22 @@ def find_target
           end
 
           binds = AssociationScope.get_bind_values(owner, reflection.chain)
-          sc.execute(binds, klass.connection) do |record|
+          records = sc.execute(binds, klass.connection) do |record|
             set_inverse_instance(record)
             if owner.strict_loading_n_plus_one_only? && reflection.macro == :has_many
               record.strict_loading!
             else
               record.strict_loading!(false, mode: owner.strict_loading_mode)
             end
           end
+
+          records.each do |record|
```

Comment:
I'd extract this out to a new method since this is not something to find the target.